tests/postgres_tests/__init__.py

```diff
@@ -12,14 +12,14 @@ class PostgreSQLTestCase(TestCase):
     @classmethod
     def tearDownClass(cls):
         # No need to keep that signal overhead for non PostgreSQL-related tests.
-        from django.contrib.postgres.signals import register_hstore_handler
+        from django.contrib.postgres.signals import register_type_handlers
 
-        connection_created.disconnect(register_hstore_handler)
+        connection_created.disconnect(register_type_handlers)
         super().tearDownClass()
 
 
 @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific tests")
 # To locate the widget's template.
 @modify_settings(INSTALLED_APPS={'append': 'django.contrib.postgres'})
-class PostgreSQLWidgetTestCase(WidgetTest):
+class PostgreSQLWidgetTestCase(WidgetTest, PostgreSQLTestCase):
```

Comment:
Makes sense. Ideally we'd move the `@modify_settings(INSTALLED_APPS={'append': 'django.contrib.postgres'})` to this mixin as well (and remove it from all it's subclasses) as we make so guarantee `'django.contrib.postgres'` objects will work without it being installed. Something along these lines should be perfect to:

```python
@unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific tests")
@modify_settings(INSTALLED_APPS={'append': 'django.contrib.postgres'})
class PostgreSQLSimpleTestCase(SimpleTestCase):
    @classmethod
    def tearDownClass(cls):
        # No need to keep that signal overhead for non PostgreSQL-related tests.
        from django.contrib.postgres.signals import register_type_handlers
        connection_created.disconnect(register_type_handlers)
        super().tearDownClass()

class PostgreSQLTestCase(PostgreSQLSimpleTestCase, TestCase):
    pass

class PostgreSQLWidgetTestCase(WidgetTest, PostgreSQLSimpleTestCase):
    pass
```